server/src/server/gamestate.rs

```diff
@@ -183,4 +185,32 @@ impl GameState {
             _ => panic!("Unprocessed message! {}", message.to_string()),
         }
     }
+
+    fn random_free_spot<R: Rng>(&self, rng: &mut R) -> (f32, f32) {
+        loop {
```

Comment:
Needs some limit on iteration count, so we don't loop endlessly in the worst case.